File: src/main/java/cc/mirukuneko/realtrainmodrenewed/client/renderer/LegacyEntityRenderState.java
package cc.mirukuneko.realtrainmodrenewed.client.renderer;

import net.minecraft.client.renderer.entity.state.EntityRenderState;
import net.minecraft.world.entity.Entity;

public final class LegacyEntityRenderState<T extends Entity> extends EntityRenderState {
    T entity;
    float entityYaw;
}

File: src/main/kotlin/cc/mirukuneko/realtrainmodrenewed/client/renderer/LegacyEntityRenderState.kt
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright © 2026 mirukuneko and RealTrainModRenewed contributors
package cc.mirukuneko.realtrainmodrenewed.client.renderer

import net.minecraft.client.renderer.entity.state.EntityRenderState
import net.minecraft.world.entity.Entity

class LegacyEntityRenderState<T : Entity> : EntityRenderState() {
    @JvmField var entity: T? = null
    @JvmField var entityYaw: Float = 0f
}
